rclcpp/src/rclcpp/subscription.cpp

```diff
@@ -80,6 +81,12 @@ SubscriptionBase::SubscriptionBase(
 
 SubscriptionBase::~SubscriptionBase()
 {
+  auto ipm = weak_ipm_.lock();
+  if (!ipm) {
+    // TODO(ivanpauno): should this raise an error?
```

Comment:
@ivanpauno consider having a flag for subscribers too to decide whether we should throw or not.